src/tests/integration/SparkLend.t.sol

```diff
@@ -0,0 +1,414 @@
+        //the outstanding unbacked aTokens minted through the bridging feature
+        uint128 unbacked;
+        //the outstanding debt borrowed against this asset in isolation mode
+        uint128 isolationModeTotalDebt;
+    }
+
+    function ADDRESSES_PROVIDER() external view returns (PoolAddressProviderLike);
+    function getReserveData(address asset) external view returns (ReserveData memory);
+    function supply(address asset, uint256 amount, address onBehalfOf, uint16 referralCode) external;
+    function borrow(address asset, uint256 amount, uint256 interestRateMode, uint16 referralCode, address onBehalfOf) external;
+    function repay(address asset, uint256 amount, uint256 rateMode, address onBehalfOf) external;
+    function withdraw(address asset, uint256 amount, address to) external;
+    function flashLoanSimple(
+        address receiverAddress,
+        address asset,
+        uint256 amount,
+        bytes calldata params,
+        uint16 referralCode
+    ) external;
+    function mintToTreasury(address[] calldata assets) external;
+}
+
+interface PoolAddressProviderLike {
+    function getPoolConfigurator() external view returns (PoolConfiguratorLike);
+}
+
+interface PoolConfiguratorLike {
+    function setSupplyCap(address asset, uint256 newSupplyCap) external;
+    function setReserveFlashLoaning(address asset, bool enabled) external;
+}
+
+interface DaiInterestRateStrategyLike {
+    function recompute() external;
+    function performanceBonus() external view returns (uint256);
+}
+
+interface ATokenLike {
+    function balanceOf(address) external view returns (uint256);
+    function approve(address, uint256) external returns (bool);
+    function transfer(address, uint256) external returns (bool);
+    function decimals() external view returns (uint8);
+    function RESERVE_TREASURY_ADDRESS() external view returns (address);
+    function scaledBalanceOf(address) external view returns (uint256);
+    function scaledTotalSupply() external view returns (uint256);
+    function totalSupply() external view returns (uint256);
+    function getIncentivesController() external view returns (address);
+}
+
+interface TreasuryLike {
+    function getFundsAdmin() external view returns (TreasuryAdminLike);
+}
+
+interface TreasuryAdminLike {
+    function transfer(
+        address collector,
+        address token,
+        address recipient,
+        uint256 amount
+    ) external;
+}
+
+contract SparkLendTest is IntegrationBaseTest {
+
+    using stdJson for string;
+    using MCD for *;
+    using GodMode for *;
+    using ScriptTools for *;
+
+    PoolLike sparkPool;
+    PoolConfiguratorLike configurator;
+    DaiInterestRateStrategyLike daiInterestRateStrategy;
+    ATokenLike adai;
+    TreasuryLike treasury;
+    TreasuryAdminLike treasuryAdmin;
+    uint256 buffer;
+
+    D3MAaveTypeBufferPlan plan;
+    D3MAaveV3NoSupplyCapTypePool pool;
+
+    function setUp() public {
+        baseInit("template-spark");
```

Comment:
Using the template input for running the tests seems weird. I don't know a better solution though, so this is probably fine for now but we should think more about this.

Reply:
Ya agree. I didn't like it either, but couldn't think of a better solution.

Reply:
Btw I think using the same `ilk` than then we will have in production will make the tests to fail once this is in production.

Reply:
Actually looking at it a bit more, we can probably hardcode values here in the tests for the config and we can pass the necessary ones to `baseInit`.
I'd go for that option than using a file for this.

Reply:
Yeah ok I like this better. done https://github.com/makerdao/dss-direct-deposit/pull/102/commits/8c82a0c3f0a5e7f581e7a3c7a5d66b8adfdb46a2

Reply:
Yeah much better!